src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1122 @@
+#endif
+
+#if defined(OPENVINO_STATIC_LIBRARY) || defined(__GNUC__) && (__GNUC__ < 4)
+    #define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __VA_ARGS__
+    #define OV_NODISCARD
+#else
+    #if defined(_WIN32)
+        #define OPENVINO_C_API_CALLBACK __cdecl
+        #ifdef openvino_c_EXPORTS
+            #define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __declspec(dllexport) __VA_ARGS__ __cdecl
+        #else
+            #define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __declspec(dllimport) __VA_ARGS__ __cdecl
+        #endif
+        #define OV_NODISCARD
+    #else
+        #define OPENVINO_C_API(...) OPENVINO_C_API_EXTERN __attribute__((visibility("default"))) __VA_ARGS__
+        #define OV_NODISCARD __attribute__((warn_unused_result))
+    #endif
+#endif
+
+#ifndef OPENVINO_C_API_CALLBACK
+    #define OPENVINO_C_API_CALLBACK
+#endif
+
+/**
+ * Max dimension of shape
+ */
+#define MAX_DIMENSION 8
+
+/**
+ * @struct ov_call_back_t
+ * @brief Completion callback definition about the function and args
+ */
+typedef struct{
+    void (OPENVINO_C_API_CALLBACK *callback_func)(void *args);
+    void *args;
+} ov_call_back_t;
+
+/**
+ * @struct ov_ProfilingInfo_t
+ */
+typedef struct{
+    /**
+     * @brief Defines the general status of a node.
+     */
+    enum Status {
+        NOT_RUN,        //!< A node is not executed.
+        OPTIMIZED_OUT,  //!< A node is optimized out during graph optimization phase.
+        EXECUTED        //!< A node is executed.
+    } status;
+
+    /**
+     * @brief The absolute time, in microseconds, that the node ran (in total).
+     */
+    int64_t real_time;
+
+    /**
+     * @brief The net host CPU time that the node ran.
+     */
+    int64_t cpu_time;
+
+    /**
+     * @brief Name of a node.
+     */
+    char* node_name;
+
+    /**
+     * @brief Execution type of a unit.
+     */
+    char* exec_type;
+
+    /**
+     * @brief Node type.
+     */
+    char* node_type;
+}ov_profiling_info_t;
+
+/**
+ * @struct ov_profiling_info_list_t
+ */
+typedef struct {
+    ov_profiling_info_t* profiling_infos;
+    size_t num;
+}ov_profiling_info_list_t;
+
+/**
+ * @enum ov_status_code_e
+ * @brief This enum contains codes for all possible return values of the interface functions
+ */
+typedef enum {
+    OK = 0,
+    GENERAL_ERROR = 1,
+    NOT_IMPLEMENTED = 2,
+    NETWORK_NOT_LOADED = 3,
+    PARAMETER_MISMATCH = 4,
+    NOT_FOUND = 5,
+    OUT_OF_BOUNDS = 6,
+    /*
+     * @brief exception not of std::exception derived type was thrown
+     */
+    UNEXPECTED = 7,
+    REQUEST_BUSY = 8,
+    RESULT_NOT_READY = 9,
+    NOT_ALLOCATED = 10,
+    INFER_NOT_STARTED = 11,
+    NETWORK_NOT_READ = 12,
+    INFER_CANCELLED = 13,
+    UNKNOWN_ERROR = 14,
+} ov_status_e;
+
+/**
+ * @struct ov_version_t
+ */
+typedef struct ov_version {
+    char* buildNumber;
+    char* description;
+}ov_version_t;
+
+typedef struct {
+    char* device_name;
+    char* buildNumber;
+    char* description;
+}ov_core_version_t;
```

Comment:
Why do we need two structures for version?

Reply:
One is openvino version and another is core version, which follows C++ interfaces.

Reply:
But C++ provides a interfaces for ov version and core version:

ov::get_openvino_version();
std::map<std::string, Version> Core::get_versions(const std::string& deviceName) ;

Reply:
I don't see 2 versions in C++ code. In C++ we have 2 ways to get version:
- dedicated function returning ov::Version
- Core::get_version returning the same object
So, they both operate with the same structure.